compio-io/src/write/mod.rs

```diff
@@ -0,0 +1,140 @@
+use compio_buf::{BufResult, IoBuf, IoVectoredBuf};
+
+use crate::IoResult;
+
+mod buf;
+mod ext;
+
+pub use buf::*;
+pub use ext::*;
+
+/// # AsyncWrite
+///
+/// Async write with a ownership of a buffer
+pub trait AsyncWrite {
+    /// Write some bytes from the buffer into this source and return a
+    /// [`BufResult`], consisting of the buffer and a [`usize`] indicating how
+    /// many bytes were written.
+    async fn write<T: IoBuf>(&mut self, buf: T) -> BufResult<usize, T>;
+
+    /// Like `write`, except that it write bytes from a buffer implements
+    /// [`IoVectoredBuf`] into the source.
+    async fn write_vectored<T: IoVectoredBuf>(&mut self, buf: T) -> BufResult<usize, T>;
```

Comment:
I think we should give a fallback implementation for `*_vectored`.

Reply:
Waiting for #93